jerry-core/ecma/base/ecma-helpers-value.c

```diff
@@ -219,6 +219,22 @@ ecma_is_value_integer_number (ecma_value_t value) /**< ecma value */
 } /* ecma_is_value_integer_number */
 
 /**
+ * Check if both values are integer ecma-numbers.
+ *
+ * @return true - if both values contain integer ecma-number values,
+ *         false - otherwise.
+ */
+inline bool __attr_pure___ __attr_always_inline___
+ecma_are_values_integer_numbers (ecma_value_t first_value, /**< first ecma value */
+                                 ecma_value_t second_value) /**< second ecma value */
+{
+  JERRY_STATIC_ASSERT (ECMA_DIRECT_TYPE_INTEGER_VALUE == 0,
+                       ecma_direct_type_integer_value_must_be_zero);
+
+  return ((first_value | second_value) & ECMA_DIRECT_TYPE_MASK) == ECMA_DIRECT_TYPE_INTEGER_VALUE;
```

Comment:
zherczeg is right here. `ECMA_DIRECT_TYPE_MASK` is 0b1011, while `ECMA_DIRECT_TYPE_INTEGER_VALUE` is 0b0000. If any of `first_value` or `second_value` is not of a direct type or not an integer value, then by or-ing them together, some of the bits in the mask will be 1. You can get all zeros only if both values are of a direct type and that direct type is integer.